src/lib.rs

```diff
@@ -68,13 +75,26 @@ mod structs {
                     }
                 }
 
-                /// Consume this struct and return the contained String
+                /// Consumes the iterator, returning the contained `String`.
                 pub fn into_inner(self) -> String {
                     self.s
                 }
 
                 #[delegate(self.i)]
-                /// Borrow the contained String
+                /// Views the underlying data as a subslice of the original data.
+                ///
+                /// # Example
+                ///
+                /// ```rust
+                /// # use owned_chars::{OwnedChars, OwnedCharsExt};
+                /// let mut chars: OwnedChars = String::from("abc").into_chars();
+                /// assert_eq!(chars.as_str(), "abc");
+                /// chars.next();
+                /// assert_eq!(chars.as_str(), "bc");
+                /// chars.next();
+                /// chars.next();
+                /// assert_eq!(chars.as_str(), "");
+                /// ```
                 pub fn as_str(&self) -> &str;
```

Comment:
I haven't actually used rust-analyzer yet. How does it show the doc/example for a function like this? I'm trying to decide whether it's worth it to use macro magic to fix the docs at the expense of rust-analyzer, like you said.

Reply:
I see... it's not great. rust-analyzer just skips any `#[doc = concat!(...)]` lines so it would look pretty bad.

Reply:
Alright I found a workaround though.

If we define this wrapper macro
```rust
macro_rules! document {
    ([$($d:expr),*] $b:item) => {
        $(#[doc=$d])*
        $b
    };
}
```

Then we can change `from_string` {and the others) like so (note: everything for the doc attributes has to be an `expr` so I had to add yet another macro argument which is the struct name as a string):

```rust
macro_rules! impls {
    ($owned_struct:ident, $owned_struct_name:expr, $target_struct:ident, $method: ident, $item: ty) => {
        impl $owned_struct {
            document! {
                ["Makes you a new `", $owned_struct_name, "` from the String for great good."]
                pub fn from_string(s: String) -> Self {
                    // etc
                }
            }

            // etc
        }
    };
}
```

And somewhat unexpectedly, rust-analyzer actually renders this!
![image](https://user-images.githubusercontent.com/47007/216903479-faf0ea0f-35d1-49e3-a602-c14c6196fa28.png)